enterprise/reporting/error_reporting.go

```diff
@@ -340,20 +340,18 @@ func (edr *ErrorDetailReporter) mainLoop(ctx context.Context, c types.SyncerConf
 				continue
 			}
 			deleteReportsStart := time.Now()
-			var delRows *sql.Rows
-			delRows, err = dbHandle.Query(`DELETE FROM `+ErrorDetailReportsTable+` WHERE reported_at = $1`, reportedAt)
+			_, err = dbHandle.ExecContext(ctx, `DELETE FROM `+ErrorDetailReportsTable+` WHERE reported_at = $1`, reportedAt)
 			errorDetailReportsDeleteQueryTimer.Since(deleteReportsStart)
 			if err != nil {
-				edr.log.Errorf(`[ Error Detail Reporting ]: Error deleting local reports from %s: %v`, ErrorDetailReportsTable, err)
+				edr.log.Errorf("[ Error Detail Reporting ]: Error deleting local reports from %s: %v", ErrorDetailReportsTable, err)
 			}
-			delRows.Close()
-		}
 
-		mainLoopTimer.Since(loopStart)
-		select {
-		case <-ctx.Done():
-			return
-		case <-time.After(edr.mainLoopSleepInterval.Load()):
+			mainLoopTimer.Since(loopStart)
+			select {
+			case <-ctx.Done():
+				return
+			case <-time.After(edr.mainLoopSleepInterval.Load()):
+			}
```

Comment:
any reason why we moved this inside the if clause?

Reply:
can we revert?

Reply:
sure.

Reply:
It still seems to be inside the if clause

Reply:
enterprise/reporting/error_reporting.go:344:33: rows.Err must be checked (rowserrcheck)
                        delRows, err = dbHandle.Query(`DELETE FROM `+ErrorDetailReportsTable+` WHERE reported_at = $1`, reportedAt)
                                                     ^
enterprise/reporting/error_reporting.go:399:28: rows.Err must be checked (rowserrcheck)
        rows, err = dbHandle.Query(`SELECT `+edSelColumns+` FROM `+ErrorDetailReportsTable+` WHERE reported_at = $1`, queryMin.Int64)
        
  was showing these lint issues

Reply:
@pras-stack maybe I was not clear. I am asking the **curly brace in line 355 to be moved to line 348** so that the following code is moved outside the `if err == nil` condition.

```go
mainLoopTimer.Since(loopStart)
select {
case <-ctx.Done():
	return
case <-time.After(edr.mainLoopSleepInterval.Load()):
}
```

Reply:
Done.